internal/appfunction/context.go

```diff
@@ -212,5 +215,30 @@ func (appContext *Context) CommandClient() interfaces.CommandClient {
 // NotificationsClient returns the Notifications client, which may be nil, from the dependency injection container
 func (appContext *Context) NotificationsClient() notifications.NotificationsClient {
 	return container.NotificationsClientFrom(appContext.dic.Get)
+}
+
+// AddValue stores a value for access within other functions in pipeline
+func (appContext *Context) AddValue(key string, value string) {
+	appContext.contextData[strings.ToLower(key)] = value
+}
+
+// RemoveValue deletes a value stored in the context at the given key
+func (appContext *Context) RemoveValue(key string) {
+	delete(appContext.contextData, strings.ToLower(key))
```

Comment:
What happens if the key doesn't exist? Should add this as a unit test

Reply:
This still not addressed in latest commit.

Reply:
sorry about that should be good now